### ISSUE TITLE
[Feature] Allow dandi organize to force usage of session_id (NWB Assets)

### ISSUE THREAD

**Issue by CodyCBakerPhD** [dandi/dandi-cli#1000]
Hey all, I'd like to be able to use `dandi organize` to force usage of both subject ID and session ID when generating a unique filename.

*E.g.*, I want to get `sub-subj1_ses-foobar_ecephys.nwb` but all I get is after standard `dandi organize` is `sub-subj1_ecephys.nwb`

This came about because I am iteratively uploading only a single NWBFile at a time to a dandiset that is being downloaded lazily (`dandi download ... --download dandiset.yaml`) in order to reduce the amount of time my very large files spend on the processing server.

**Comment by yarikoptic** [comment]
Yet to check in detail, but I think it is at large duplicate of #69

**Comment by CodyCBakerPhD** [comment]
@yarikoptic Specifically the last comment by @satra in that thread (https://github.com/dandi/dandi-cli/issues/69#issuecomment-833629364, the part about 'partial access' to data) kind of touches on a part of why I'm requesting this...

But the issue would still exist for the very first session of each subject I upload, which acts as if the subject ID alone is sufficient to be 'unique', but really it needs the full session ID added on for that. But I won't have two files present at time of organize to force the automated usage of the session ID, so a manual control for that would be nice.

**Comment by waxlamp** [comment]
@AlmightyYakob and I (with @CodyCBakerPhD's help) discovered that this may be leading to bad behavior in general: apparently, repeated upload of a disambiguated filename of the kind Cody is talking about here led to multiple assets being stored in the dandiset at the exact same pathname. This resulted in two assets in the dandiset only one of which was visible in the file browser.

We have manually renamed the later version of the duplicate pathname and communicated with Cory about it, and we expect that under the new AssetPath work in the archive that this shouldn't happen again, but it may instead result in an error about the target file already existing in the dandiset. We may find out soon as Cody's CI runs and uploads files to that folder again.

**Comment by satra** [comment]
the place where this gets tricky is partial uploads and perhaps where @CodyCBakerPhD is running into an issue.

`dandi organize` currently assumes all your data are local for path disambiguation. there is an open issue i think to support partial dataset disambiguation, i.e. take into account both local and remote to do disambiguation and also change paths of remote assets as a result. this has not been implemented.

**Comment by waxlamp** [comment]
I don't know specifically what caused multiple assets to be coexisting in the database with identical pathnames. Perhaps it was a dandi-archive bug, or perhaps it was intended behavior, with a future GC capability intended to remove orphaned versions. This came up with @AlmightyYakob was applying the new AssetPath checks and Cody's dandiset was the one place it came up. If the duplicate asset path doesn't appear again during regular use of Cody's CI, then I believe we can chalk it up to something that was buggy and no longer is; if we run into difficulties, we can decide at that point what the proper behavior of the archive ought to be, and adjust the logic to account for that.

That is to say, no action needed at the moment nor any bugs to address, and we're just waiting on more data from the CI outcomes.

**Comment by yarikoptic** [comment]
I stay with opinion that it is at large a duplicate of #69 .
